app/src/test/java/org/mozilla/fenix/components/InflationAwareFeatureTest.kt

```diff
@@ -13,6 +13,11 @@ import org.mockito.Mockito.verify
 import java.lang.ref.WeakReference
 
 class InflationAwareFeatureTest {
+
+    private val stub: ViewStub = mock()
+    private val inflationFeature: InflationAwareFeature = spy(TestableInflationAwareFeature(stub))
+    private val innerFeature: LifecycleAwareFeature = mock()
```

Comment:
Reusing spys and mocks across tests can lead to a bunch of very hard to debug issues and wrong assertions.

We could use `lateinit var` and set them up from a `@Before` method for every test. But tbh I'm not sure whether its worth is. Explicit test isolation by duplication is not necessarily bad.

Reply:
Yes, I understand your point. I actually ran the tests and all of them were passing. Since the spies were created the same way again I decided to extract them out of the functions. If you'd like me to change it back to the way it was before (each function creates a spy), let me know. However, using `lateinit var` and `@Before` would result in the same thing I guess. And I opted for `val` to keep the code as consistent as possible.

Also, any other suggestions?

Reply:
@usmanrafi I think using `@Before` and `private lateinit var` is the pattern I'd suggest here. It's what we're using in most of our new tests and what I was suggesting in #4556.

Reply:
@sblatz Okay, I'll refactor the tests again. What do you think about the comment by @pocmo ? I think reusing mocks is the better option (only in the case that same mocks are being used and created in the same way)

Reply:
Yep, agreed.

Reply:
> Reusing spys and mocks across tests can lead to a bunch of very hard to debug issues and wrong assertions.
> We could use lateinit var and set them up from a @Before method for every test.

I actually don't think there's a functional difference between initializing the test member variables at the declaration site vs initializing in `@Before` [since a fresh instance of the class is created for each test method](http://junit.sourceforge.net/javadoc/org/junit/Test.html).